src/cephadm/cephadm

```diff
@@ -2164,16 +2307,18 @@ class Firewalld(object):
         call_throws([self.cmd, '--reload'])
 
 
-def update_firewalld(daemon_type):
-    # type: (str) -> None
+def update_firewalld(fsid: str, daemon_type: str,
+                     daemon_id: Union[int, str]) -> None:
     firewall = Firewalld()
-
     firewall.enable_service_for(daemon_type)
 
     fw_ports = []
 
     if daemon_type in Monitoring.port_map.keys():
         fw_ports.extend(Monitoring.port_map[daemon_type])  # prometheus etc
+    elif daemon_type == CustomContainer.daemon_type:
+        cc = CustomContainer.init(fsid, daemon_id)
```

Comment:
In my opinion it does not seem very logical to include the "fsid" in the "update_firewall" function, because open or not ports is something directly linked with the daemon configuration.( not with the type of daemon)
What happens if i want to create a CustomContainer, should I call to "update_firewall"? does not seem very intuitive....
But in ant case the problem is in the function itself, because is not operative to open ports taking into account just the type of daemon. (I proposed a refactor of this to avoid this kind of things) 
We will need to return over this to change completely the way we are managing firewall.

Reply:
> I proposed a refactor of this to avoid this kind of things

Good idea, but not as part of this PR. IMO there is much more to refactor in the `cephadm` code base. The intention of this PR is to keep the code changes as low as possible.

Reply:
After looking into the code again i realized that there is some code duplication. I do not need to enhance `update_firewalld` anymore.